api/logic/Explanation.java

```diff
@@ -30,7 +31,7 @@ public interface Explanation {
 
     Rule rule();
 
-    ConceptMap conclusion();
+    Map<String, Concept> conclusion();
```

Comment:
I'm a little confused. A `ConceptMap` **is** a `Map<String, Concept>`. What is the distinction here?

Reply:
Yes I see the confusion here, there are two distinctions.
1. A `ConceptMap` can contain only retrievable variables on the server, whereas a `conclusion` can have non-retrievable variables. So, the actual variable names that can occur in a `conclusion` is a superset of those that can occur in a `ConceptMap`.
2. A `ConceptMap` has `Explainables`. We never have this for a `Conclusion`, so we have just been leaving this empty so far when using a `ConceptMap`.

The change then is due to the above given that there is a semantic difference between the two. This is also less confusing for us devs, because if we used `ConceptMap` for `conclusion` here then it would be inconsistent with the server. The user would also see an `Explainables` that is always empty.